Updating Verfified -> Partner (#115)
hashicorp/terraform-docs-common: website/docs/registry/providers/index.mdx

<ProviderTable />
<p></p>

## Partner Provider Development Program

> If your organization is interested in joining our Provider Development Program (which sets the standards for publishing providers and modules with a `Partner` badge), please take a look at our [Program Details](/docs/partnerships) for further information.

Reviewer: ```suggestion
If your organization wants to join our Provider Development Program (which sets the standards for publishing providers and modules with a `Partner` badge), please take a look at our [Program Details](/docs/partnerships) for further information.
```

Suggest changing "is interested in joining" to "wants to join" to avoid passive voice style issues.
